test/Pool.test.js

```diff
@@ -1315,6 +1315,22 @@ describe('Pool', function () {
     });
   });
 
+  it('should not terminate worker if abort listener is defined', function () {
```

Comment:
I think we can use some additional unit tests:

1. test that checks that the second task does not trigger the abort listener of the previous task (ie. verify that the listener is cleaned up)
2. test the ability to attach multiple abort listeners (for example 2)
3. test the behavior when the abort listener never resolves
4. verify that you can use an abort listener in an offloaded function (not a dedicated worker)

Does that make sense? More ideas?

Reply:
I think these are good surface area. Do we need to update any existing tests?

Reply:
I think all existing tests can be in place: this will not be a breaking change in behavior, only a new feature, so we should not need to alter existing tests.